### PR TITLE
Show files from global announcements in a web view

### PR DESCRIPTION
**Pull request by armstrongnate**
refs: MBL-14931
affects: student, teacher
release note: Fixed viewing files linked in global announcements

test plan: see ticket

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  One or more files are below the minimum test coverage 50%
  </td>
    </tr>
  </tbody>
</table>


Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 92.11% | -- | --
Core/Core/Dashboard/DashboardInviteSection.swift | 0% | -- | --
Core/Core/Submissions/GradeSubmission.swift | 0% | -- | --
Core/Core/Dashboard/DashboardCourseSection.swift | 36.67% | -- | --
Core/Core/Dashboard/DashboardConferenceSection.swift | 0% | -- | --
Core/Core/Extensions/TransactionExtensions.swift | 0% | -- | --
Core/Core/Dashboard/DashboardAnnouncementSection.swift | 0% | -- | --

[MBL-14931](https://instructure.atlassian.net/browse/MBL-14931)

#### Affected Apps: Student, Teacher

#### Release Note: 
Fixed viewing files linked in global announcements
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by inst-danger** [comment]
<details><summary>Student.ipa</summary>
  <a href="https://app.bitrise.io/artifact/55931170/p/d534ab522de4414c57173a73d418061c">
    
  </a>
</details>
<details><summary>Teacher.ipa</summary>
  <a href="https://app.bitrise.io/artifact/55931184/p/15d4de3a2d3ab88b06d6d08b69e377c4">
    
  </a>
</details>
<details><summary>Parent.ipa</summary>
  <a href="https://app.bitrise.io/artifact/55931130/p/2b823252092c2c690d084b2a9c2fc2cb">
    
  </a>
</details>

**Review by vanwagonet — APPROVED** [review]
QA+1

File: rn/Teacher/ios/Teacher/Routes.swift [hunk 1]
```diff
@@ -317,6 +317,13 @@ private func fileList(url: URLComponents, params: [String: String], userInfo: [S
 }
 
 private func fileDetails(url: URLComponents, params: [String: String], userInfo: [String: Any]?) -> UIViewController? {
+    // Show files from global announcements in a web view because they are not available via the api.
+    if let urlRaw = url.url, url.queryItems?.first(where: { $0.name == "origin" })?.value == "globalAnnouncement" {
```
[the remark this answers is not in the document]
Comment (armstrongnate) [reply rn/Teacher/ios/Teacher/Routes.swift]: Good idea! Done.

**Review by vanwagonet — APPROVED** [review]
QA+1

**Pull request merged**